### PR TITLE
Debugged the nominatim function, so it runs

### PR DESCRIPTION
**Pull request by valin1**
This did not improve accuracy, but I found some tests that technically should've passed in terms of finding the correct business name, but didn't match the street address exactly. Because nominatim's street address includes the business name as well in the street address. --> Will need to address the expected results in the tests. 

Verdict: I think we should update our server, because when I tried changing the link to the public server like https://nominatim.openstreetmap.org/reverse? and the MapQuest version with our key http://open.mapquestapi.com/nominatim/v1/reverse.php?key=[our key], I got this error. 

```
Traceback (most recent call last):
  File "/Users/vanessalin/Desktop/e-mission-server/emission/net/ext_service/geocoder/nominatim.py", line 95, in reverse_geocode
    jsn = cls.get_json_reverse(lat, lng)
  File "/Users/vanessalin/Desktop/e-mission-server/emission/net/ext_service/geocoder/nominatim.py", line 87, in get_json_reverse
    response = urllib.request.urlopen(request)
  File "/Users/vanessalin/anaconda2/envs/emission/lib/python3.6/urllib/request.py", line 223, in urlopen
    return opener.open(url, data, timeout)
  File "/Users/vanessalin/anaconda2/envs/emission/lib/python3.6/urllib/request.py", line 532, in open
    response = meth(req, response)
  File "/Users/vanessalin/anaconda2/envs/emission/lib/python3.6/urllib/request.py", line 642, in http_response
    'http', request, response, code, msg, hdrs)
  File "/Users/vanessalin/anaconda2/envs/emission/lib/python3.6/urllib/request.py", line 570, in error
    return self._call_chain(*args)
  File "/Users/vanessalin/anaconda2/envs/emission/lib/python3.6/urllib/request.py", line 504, in _call_chain
    result = func(*args)
  File "/Users/vanessalin/anaconda2/envs/emission/lib/python3.6/urllib/request.py", line 650, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
urllib.error.HTTPError: HTTP Error 403: Forbidden
```

Example of false tests: 
```
DEBUG:root:Test Mission San Jose High School Fremont failed, output = ['Mission San Jose High School', 'Mission San Jose High School, Mission Creek Trail, Mission San Jose District, Fremont, Alameda County, 94539, United States of America', 'Fremont', False], expected ['Mission San Jose High School', 'Palm Avenue', 'Fremont', True]
```
```
DEBUG:root:Test Kaiser Permanente Pediatrics Building failed, output = ['Kaiser Permanente Pediatrics Building', 'Kaiser Permanente Pediatrics Building, Los Coches Street, Milpitas, Santa Clara County, 95035, United States of America', 'Milpitas', False], expected ['Kaiser Permanente Pediatrics Building', 'Los Coches Street', 'Milpitas', True]
```
```
DEBUG:root:Test Bank of America failed, output = ['Bank of America', 'Bank of America, South Hillview Drive, Milpitas, Santa Clara County, 95035, United States of America', 'Milpitas', False], expected ['Bank of America', 'East Calaveras Boulevard', 'Milpitas', True]
```
Examples of same streets, but no business names included (streetnames become business_names)
```
DEBUG:root:Test San Ramon Valley High School failed, output = ['Danville Boulevard', 'Danville Boulevard, Danville, Contra Costa County, 94526, United States of America', 'Danville', False], expected ['San Ramon Valley High School', 'Danville Boulevard', 'San Ramon', True]
```

### CODE REVIEW

**Comment by shankari** [comment]
@valin1 I agree that the false tests should be fixed, but should the "streetnames become business names" really be called correct? If you then look up that business name, you are not going to find a match on yelp, right?

File: emission/core/wrapper/suggestion_sys.py [hunk 1]
```diff
@@ -200,16 +208,16 @@ def find_destination_business(lat, lon):
     # print(return_address_from_location_google(location))
     # print(len(return_address_from_location_google(location)))
     #IF RETURN_ADDRESS_FROM_LOCATION HAS A BUSINESS LOCATION ATTACHED TO THE ADDRESS
-    try:
+    # try:
```
Comment (shankari) [inline emission/core/wrapper/suggestion_sys.py]: why is this commented out?

**Comment by shankari** [comment]
> Verdict: I think we should update our server, because when I tried changing the link to the public 
> server like https://nominatim.openstreetmap.org/reverse? and the MapQuest version with our key 
> http://open.mapquestapi.com/nominatim/v1/reverse.php?key=[our key], I got this error.

I think you should debug this further wrt the query that you are using. Are you telling me that OSM and MapQuest are broken? I think not. Maybe the query format also changed subtly in newer versions?

**Comment by shankari** [comment]
wrt the street addresses, yes, we knew that the check for `find_business_for_name` was unlikely to be perfect. I guess I could change the harness for this case to only check the name instead of all the fields.

**Comment by shankari** [comment]
I am 100% convinced that the nominatim services are not broken because this works
https://nominatim.openstreetmap.org/reverse.php?format=html&lat=37.40355&lon=-122.10783&zoom=18
and so does this
https://nominatim.openstreetmap.org/reverse.php?format=json&lat=37.40355&lon=-122.10783&zoom=18

**Comment by valin1** [comment]
```
INFO:root:Testing candidate find_destination_business complete, overall accuracy = 7.317073170731708 
INFO:root:Test complete, comparison results = 
INFO:root:candidate: find_destination_business, accuracy = 7.317073170731708
``` 

Lower percentage than before, this is using the current server that we have.

**Comment by shankari** [comment]
@valin1 if you look at the example, expected results no longer have "True" in them; hopefully no business name is `True`. Although given https://github.com/e-mission/e-mission-docs/issues/388 maybe we should keep it in? Although I think that determining whether a place is a service or not is actually its own step, since it is a heuristic that you came up with and other heuristics are possible.

File: emission/core/wrapper/suggestion_sys.py (hunk 1 above)
Comment (shankari) [reply emission/core/wrapper/suggestion_sys.py]: @valin1 no response to this, code still in place?!

Comment (valin1) [reply emission/core/wrapper/suggestion_sys.py]: oh this was commented out bc i thought it was defaulting to google api

Comment (valin1) [reply emission/core/wrapper/suggestion_sys.py]: i will uncomment

**Comment by valin1** [comment]
Accuracy 
```
DEBUG:root:For candidate find_destination_business, after instance Department of Motor Vehicles, successfulTests = 3, failedTests = 38
INFO:root:Testing candidate find_destination_business complete, overall accuracy = 7.317073170731708 
INFO:root:Test complete, comparison results = 
INFO:root:candidate: find_destination_business, accuracy = 7.317073170731708
```

File: emission/integrationTests/suggestionsys/find_destination_business.dataset.json [hunk 2]
```diff
@@ -19,7 +19,7 @@
                 "type": "Point"
             }
         },
-        "output": ["Kettle", "North Highland Avenue", true]
```
Comment (shankari) [inline emission/integrationTests/suggestionsys/find_destination_business.dataset.json]: we don't know that the other business names will follow this format.
I checked the name for Kohl's in OSM and yelp and put them in.
There is no guarantee that yelp follows the same format for other businesses and in fact it does not for "Kettle"

> if people find alternate spellings for them while evaluating google/yelp, they should change the datasets as part of their commit.

either you do that or you actually find the names in the other services and put them in. Doesn't make sense to put in random names

Comment (valin1) [reply emission/integrationTests/suggestionsys/find_destination_business.dataset.json]: Resolved

**Comment by valin1** [comment]
```
INFO:root:Testing candidate find_destination_business complete, overall accuracy = 7.317073170731708 
INFO:root:Test complete, comparison results = 
INFO:root:candidate: find_destination_business, accuracy = 7.317073170731708
```

**Comment by valin1** [comment]
```
DEBUG:root:For candidate find_destination_business, after instance Department of Motor Vehicles, successfulTests = 1, failedTests = 40
INFO:root:Testing candidate find_destination_business complete, overall accuracy = 2.4390243902439024 
INFO:root:Test complete, comparison results = 
INFO:root:candidate: find_destination_business, accuracy = 2.4390243902439024
``` 
From changing the query url.

**Comment by valin1** [comment]
```
urllib.error.HTTPError: HTTP Error 403: Forbidden
```
403 Error was resolved by using "query_url" : "https://nominatim.openstreetmap.org/reverse.php?format=json&" 

Original error was due to the incorrect formatting of the URL.

**Comment by shankari** [comment]
@valin1 can you also indicate how you got the data to update the dataset file?

**Comment by valin1** [comment]
I got the data through searching for the business on Yelp and using the business ID with sugg.request. 
Emma said that there was one test that should've been marked successful in the dataset. 

```
DEBUG:root:For amenity Chicago, comparing address 111 South Michigan Avenue, Chicago with nearby business Chicago
DEBUG:root:For amenity The Art Institute of Chicago, comparing address 111 South Michigan Avenue, Chicago with nearby business 111 South Michigan Avenue, Chicago
DEBUG:root:After checking = 111 South Michigan Avenue, Chicago, got business tuple (True, 'The Art Institute of Chicago') 
DEBUG:root:no match! name = The Art Institute of Chicago, comparing with ["Art Institute's Gardens", 'Art Institute of Chicago Chicago 3']
DEBUG:root:Test Art Institute of Chicago Garden failed, output = False, expected ["Art Institute's Gardens", 'Art Institute of Chicago Chicago 3'] 
DEBUG:root:For candidate find_destination_business_google, after instance Art Institute of Chicago Garden, successfulTests = 1, failedTests = 29
```

**Pull request merged**